stl/inc/xstring

```diff
@@ -2642,6 +2655,50 @@ private:
     }
 
 public:
+#if _HAS_CXX20
+    void _Move_assign_from_buffer(_Elem* const _Right, const size_type _Size, const size_type _Res) {
+        // Move assign from a buffer, used by basic_stringbuf
+        _Tidy_deallocate();
+        _Traits::assign(_Right[_Size], _Elem());
+        pointer _Fancy_right = pointer_traits<pointer>::pointer_to(*_Right);
+        auto& _My_data       = _Mypair._Myval2;
+        _My_data._Mysize     = _Size;
+        _My_data._Myres      = _Res;
+        if (_My_data._Large_string_engaged()) {
+            _Construct_in_place(_My_data._Bx._Ptr, _Fancy_right);
+        } else {
+            _Traits::copy(_My_data._Bx._Buf, _Right, _Res);
+            _Getal().deallocate(_Fancy_right, _Res);
+        }
+    }
+
+    struct _Released_buffer {
+        pointer _Ptr;
+        size_type _Size;
+        size_type _Res;
+    };
+
+    _NODISCARD _Released_buffer _Release_to_buffer(_Alloc& _Al) {
+        // Release to a buffer, or allocate a new one if in small string mode
+        auto& _My_data = _Mypair._Myval2;
+        if (_My_data._Large_string_engaged()) {
+            _Released_buffer _Result{_My_data._Bx._Ptr, _My_data._Mysize, _My_data._Myres};
+            _Tidy_leak();
+            return _Result;
+        } else {
+            // use _BUF_SIZE + 1 to avoid SSO, if the buffer is assigned back
+            _Released_buffer _Result{_Al.allocate(_BUF_SIZE + 1), _My_data._Mysize, _BUF_SIZE + 1};
+            _Traits::copy(_Unfancy(_Result._Ptr), _My_data._Bx._Buf, _BUF_SIZE);
+            _Tidy_leak();
+            return _Result;
+        }
+    }
+
+    _NODISCARD _Alty& _Get_allocator_reference() {

```

Comment:
I think you should `friend` the place that needs to get this and call `_Getal()` rather than introducing another function for it (or just call `get_allocator()`)